### PR TITLE
feat: Sardine

### PR DESCRIPTION
**Pull request by mahume**
- add methods and API endpoints for fetching supported tokens from Sardine
- add tests for new function

### CODE REVIEW

File: packages/bitcore-wallet-service/src/lib/expressapp.ts [hunk 1]
```diff
@@ -1745,6 +1745,19 @@ export class ExpressApp {
       });
     });
 
+    router.post('/v1/service/sardine/getSupportedTokens', (req, res) => {
+      getServerWithAuth(req, res, server => {
```
Comment (Gamboster) [inline packages/bitcore-wallet-service/src/lib/expressapp.ts:1749]: I would not use getServerWithAuth to fetch supported tokens/currencies. You can do it directly like this example:

router.post('/v1/service/ramp/assets', async (req, res) => {
      let server, response;
      try {
        server = getServer(req, res);
        response = await server.rampGetAssets(req);
        return res.json(response);
      } catch (err) {
        return returnError(err ?? 'unknown', res, req);
      }
    });

File: packages/bitcore-wallet-client/src/lib/api.ts [hunk 2]
```diff
@@ -3415,6 +3415,15 @@ export class API extends EventEmitter {
     });
   }
 
+  sardineGetSupportedTokens(data): Promise<any> {
```
Comment (Gamboster) [inline packages/bitcore-wallet-client/src/lib/api.ts:3418]: A BWC bump will not be necessary. By not using getServerWithAuth we can access through our BWS URL directly (.../v1/service/sardine/getSupportedTokens). So you can remove this function.

File: packages/bitcore-wallet-service/src/lib/server.ts [hunk 3]
```diff
@@ -5787,6 +5787,41 @@ export class WalletService implements IWalletService {
...
+      const secretBase64 = Buffer.from(secret).toString('base64');
+
+      const headers = {
+        'Content-Type': 'application/json',
+        'Authorization': `Basic ${secretBase64}`,
+      };
+
+      const URL: string = API + '/v1/supported-tokens';
+
+      this.request.post(
```
Comment (Gamboster) [inline packages/bitcore-wallet-service/src/lib/server.ts:5807]: we should use this.request.get(...

File: packages/bitcore-wallet-service/src/lib/server.ts [hunk 4]
```diff
@@ -5787,6 +5787,41 @@ export class WalletService implements IWalletService {
...
+        URL,
+        {
+          headers,
+          body: req.body,
```
Comment (Gamboster) [inline packages/bitcore-wallet-service/src/lib/server.ts:5811]: This method does not requiere body parameters
https://docs.sardine.ai/docs/integrate-payments/on-ramp-api/operations/list-supported-tokens

**Review by Gamboster — CHANGES_REQUESTED** [review]

**Comment by mahume** [comment]
Changes pushed. Please test locally as I am coding this blind without Bitcore running locally.

**Review by Gamboster — APPROVED** [review]
LGTM

**Review by kajoseph — APPROVED** [review]

**Pull request merged**